src/app/(admin)/new_dashboard/hooks/useChat.ts

```diff
@@ -1,41 +1,77 @@
-import { useState } from "react";
-import { Message } from "../types";
+import { useEffect, useState } from "react";
+import { Chat, Message } from "@/types";
+import socket from "@/socket";
+import { useChatById } from "@/hooks/api/chatHooks";
+import { SuccessResponse } from "@/api/base.client";
 
-export const useChat = () => {
-  // Initial dummy messages
-  const initialMessages: Message[] = [
-    { sender: "User 1", text: "Hello!", timestamp: "10:00 AM" },
-    { sender: "User 2", text: "Hi there!", timestamp: "10:01 AM" },
-  ];
-
-  const [messages, setMessages] = useState<Message[]>(initialMessages);
+export const useChat = (chatId: number | null) => {
+  const [chatMessages, setChatMessages] = useState<Message[]>([]);
   const [newMessage, setNewMessage] = useState<string>("");
+  const [isNote, setIsNote] = useState<boolean>(false);
+
+  const { data, error, isError } = useChatById(chatId);
+  useEffect(() => {
+    if (chatId) {
+      socket.emit("join", { chatId });
+
+      socket.on("newMessage", (message: Message) => {
+        setChatMessages((prevMessages) => [...prevMessages, message]);
+      });
+
+      return () => {
+        socket.off("newMessage");
+        socket.emit("leave", { chatId });
+      };
+    }
+  }, [chatId]);
+
+  useEffect(() => {
+    if (isError) {
+      console.error(error);
+    }
+    if (data) {
+      const { data: chat } = data as SuccessResponse<Chat>;
+      const messages = chat.messages as Message[];
+      setChatMessages(messages);
+    }
+  }, [data, error, isError]);
 
   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.value.split("\n").length > 10) return;
```

Comment:
I recommend using react hook form and zod validation for all of this hook. It would be much easier for you to validate the types and reset the form messages as you need.